### PR TITLE
[flang] NFC Fix out-of-tree builds.

### PR DESCRIPTION
**Pull request by psteinfeld**
The relative paths were only working for in-tree builds.

### CODE REVIEW

File: flang/lib/Optimizer/Builder/Runtime/Assign.cpp [hunk 1]
```diff
@@ -7,7 +7,7 @@
 //===----------------------------------------------------------------------===//
 
 #include "flang/Optimizer/Builder/Runtime/Assign.h"
-#include "../../../runtime/assign.h"
+#include "flang/../../../flang/runtime/assign.h"
```
Comment (schweitzpgi) [inline flang/lib/Optimizer/Builder/Runtime/Assign.cpp:10]: Doesn't just adding a fourth `../` fix the problem? `runtime` is a sibling of `lib` after all.

**Comment by jeanPerier** [comment]
I am not sure what is the best fix here. Given we need to include these headers in lowering, should we see with @klausler if they can be moved in an `flang/include/flang/Runtime` directory so that we are not dependent on the relative paths ?

File: flang/lib/Optimizer/Builder/Runtime/Assign.cpp (hunk 1 above)
Comment (psteinfeld) [reply flang/lib/Optimizer/Builder/Runtime/Assign.cpp:10]: Yes.  Adding a fourth `../` works too.  I personally find it less confusing to use `flang/`.

Comment (schweitzpgi) [reply flang/lib/Optimizer/Builder/Runtime/Assign.cpp:10]: Actually, I find it more confusing. It's using the `-I` search path to find the `.../flang/include/flang/` path, then backing up to `.../` then adding the "other" `flang` directory (at the top-level). The preexisting notion was to just backtrack from the directory in which this `.cpp` file is located to the right header file.

I better solution overall would be to have the runtime header files in the include directory like the other exported headers, of course.

**Comment by schweitzpgi** [comment]
Should this be closed now?

**Comment by psteinfeld** [comment]
> 
> 
> Should this be closed now?

Yep.

**Comment by psteinfeld** [comment]
All problems with out-of-tree builds are now resolved.

**Pull request closed without merging**